packages/entities/entities-plugins/src/components/free-form/Datakit/flow-editor/FlowPanels.vue

```diff
@@ -0,0 +1,321 @@
+const responseFlowStore = useVueFlow(responseFlowId)
+
+const flowPanels = useTemplateRef('flowPanels')
+const requestLabel = useTemplateRef('requestLabel')
+const requestCanvasContainer = useTemplateRef('requestCanvasContainer')
+const requestFlow = useTemplateRef('requestFlow')
+const resizeHandle = useTemplateRef('resizeHandle')
+const responseLabel = useTemplateRef('responseLabel')
+const responseCanvasContainer = useTemplateRef('responseCanvasContainer')
+const responseFlow = useTemplateRef('responseFlow')
+
+const flowPanelsRect = useElementBounding(flowPanels)
+const requestLabelRect = useElementBounding(requestLabel)
+const requestCanvasContainerRect = useElementBounding(requestCanvasContainer)
+const resizeHandleRect = useElementBounding(resizeHandle)
+const responseLabelRect = useElementBounding(responseLabel)
+const responseCanvasContainerRect = useElementBounding(responseCanvasContainer)
+
+const snappingHeight = computed(() => Math.max(30, flowPanelsRect.height.value * 0.03))
+const isRequestCollapsed = computed(() => requestCanvasContainerRect.height.value < snappingHeight.value)
+const isResponseCollapsed = computed(() => responseCanvasContainerRect.height.value < snappingHeight.value)
+
+const requestHeight = ref('50%')
+const responseHeight = ref('50%')
+
+const onRequestLabelClick = () => {
+  if (!resizable) return
+
+  if (isRequestCollapsed.value) {
+    requestHeight.value = '50%'
+    responseHeight.value = '50%'
+  } else {
+    requestHeight.value = '0'
+    responseHeight.value = '100%'
+  }
+}
+
+const onResponseLabelClick = () => {
+  if (!resizable) return
+
+  if (isResponseCollapsed.value) {
+    requestHeight.value = '50%'
+    responseHeight.value = '50%'
+  } else {
+    requestHeight.value = '100%'
+    responseHeight.value = '0'
+  }
+}
+
+const requestResizableHeight = computed(() =>
+  flowPanelsRect.height.value - requestLabelRect.height.value - resizeHandleRect.height.value - responseLabelRect.height.value,
+)
+
+const { isDragging } = useDraggable(resizeHandle, {
+  onMove(position) {
+    if (!resizable) return
+
+    const handleCenterToRequestCanvasTop = Math.max(0, position.y - requestCanvasContainerRect.top.value)
+    const requestProportion = Math.min(1,
+      Math.max(0,
+        handleCenterToRequestCanvasTop / requestResizableHeight.value,
+      ),
+    )
+    requestHeight.value = `${requestProportion * 100}%`
+    responseHeight.value = `${(1 - requestProportion) * 100}%`
+  },
+  onEnd() {
+    if (!resizable) return
+
+    if (requestCanvasContainerRect.height.value < snappingHeight.value) {
+      requestHeight.value = '0'
+      responseHeight.value = '100%'
+    } else if (responseCanvasContainerRect.height.value < snappingHeight.value) {
+      requestHeight.value = '100%'
+      responseHeight.value = '0'
+    }
+  },
+})
+
+// 2-pass: Fit view and sync the zoom
+const fitView = () => {
+  // Hack. Next tick does not work as expected here
```

Comment:
What issues are you running into? I did some research while building a PoC and might be able to share my solution.

Reply:
I previously found that although the two flows were fitted to view on init, upon a click on the "Fit View" button, the viewport moved slightly, which indicates the layout computation is not yet finalised by the point when the initial fit view happened.

However, since we no longer need to sync the zoom level between both flows. A simple `nextTick` worked here again.